component/file_cache/file_cache.go

```diff
@@ -890,8 +894,11 @@ func (fc *FileCache) ReadFile(options internal.ReadFileOptions) ([]byte, error)
 func (fc *FileCache) ReadInBuffer(options internal.ReadInBufferOptions) (int, error) {
 	//defer exectime.StatTimeCurrentBlock("FileCache::ReadInBuffer")()
 	// The file should already be in the cache since CreateFile/OpenFile was called before and a shared lock was acquired.
-	localPath := filepath.Join(fc.tmpPath, options.Handle.Path)
-	fc.policy.CacheValid(localPath)
+	options.Handle.OptCnt++
+	if (options.Handle.OptCnt % 1000) == 0 {
```

Comment:
Should we add a comment here about why we are only doing this every 1000 calls?

Reply:
As such there is no reason for selecting the value 1K. But updating cache invovles locking and all so doing it on every call has its own cost. Also post this PR default will be to do read/write from native code and there for read m not updating cache at all for write we will update every 1K calls. here 1K calls translates to 128K * 1000 ~= 128MB so every 128MB read we will update the cache which shall ideally be well over few seconds.

Reply:
Can we add this somewhere in the source as a comment?